### PR TITLE
Allow sorted attribute names for the amp-ad-metadata script.

### PR DESCRIPTION
**Pull request by powdercloud**
And look for that variant first.

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -1498,13 +1502,18 @@ export class AmpA4A extends AMP.BaseElement {
    * TODO(keithwrightbos@): report error cases
    */
   getAmpAdMetadata_(creative) {
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: This entire search can be simplified with a regex:

```js
const regex = /(<script +type=(['"])?application\/json\2 +amp-ad-metadata *>|<script +amp-ad-metadata +type=(['"])?application\/json\3 *>)(.*?)<\/script>/;

{
  getAmpAdMetadata_(creative) {
    const match = regex.exec(creative);
    if (!match) {
      // warn
      return null;
    }

    try {
      const metaDataObj = JSON.parse(match[4]);
      // ...
    } catch (e) {
      // ...
    }
  }
}

Comment (powdercloud) [inline extensions/amp-a4a/0.1/amp-a4a.js]: @jridgewell I'd like to get rid of all these variants though. Do you think it's worth having a regex in the meantime?

Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Which variants? Aren't we adding a variant?

Comment (powdercloud) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Yes, I'm adding a variant. The idea is to make it the only variant (it's the simplest, most orderly variant), but to do this we'll need to change the server side, once this change is released, and then come back.
The reason why sorting the attribute names is useful is that we do it for everything else on the server side now. So this thing sticks out like a sore thumb, and could break by accident.

Comment (powdercloud) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Also: sorry for not making my overall intentions clearer initially. But basically I wish to simplify / straighten this, not make it more complicated, and I'm willing to suffer temporary complexity but not permanent complexity for it.

Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Sure.

**Comment by taymonbeal** [comment]
@ampproject/cloudflare, we'd like to switch to `<script amp-ad-metadata type=application/json>` as the required search string for Fast Fetch metadata. The signing service spec would specify that the metadata element's opening tag has to be that exact string, character for character. Do you have any objections to this? The other variants would continue to be accepted for now, so we wouldn't break you until you'd made the switch.

**Comment by irtefa** [comment]
@oliy thoughts on this proposal?

**Comment by oliy** [comment]
That would be a straightforward enough change. We can do that.

**Pull request closed without merging**